tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -221,22 +221,19 @@ def test_transitive_params(self):
 
     # Test that an inner Get in transitive_coroutine_rule() is able to resolve B from C due to the
     # existence of transitive_b_c().
-    result_d = self.request_single_product(D, Params(c))
-    # We don't need the inner B objects to be the same, and we know the arguments are type-checked,
-    # we're just testing transitively resolving products in this file.
-    self.assertTrue(isinstance(result_d, D))
+    with self.assertDoesNotRaise():
+      _ = self.request_single_product(D, Params(c))
 
   @contextmanager
   def _assert_execution_error(self, expected_msg):
     with assert_execution_error(self, expected_msg):
       yield
 
   def test_union_rules(self):
-    a = self.request_single_product(A, Params(UnionWrapper(UnionA())))
-    # TODO: figure out what to assert here!
-    self.assertTrue(isinstance(a, A))
-    a = self.request_single_product(A, Params(UnionWrapper(UnionB())))
-    self.assertTrue(isinstance(a, A))
+    with self.assertDoesNotRaise():
+      _ = self.request_single_product(A, Params(UnionWrapper(UnionA())))
```

Comment:
You should be able to remove `_ =` for less noise.

Reply:
So I've made the other change you suggested, but I'm seeing `_ = ` in a few other places in this file, and was thinking it seemed more appropriate to use that, even though the `_` is strictly unnecessary.